app/lib/katello/resources/cdn.rb

```diff
@@ -139,6 +141,42 @@ def parse_host(host_or_url)
           return uri.host || uri.path
         end
 
+        def get_container_listings(content_path)
+          JSON.parse(get(File.join(content_path, CdnResource::CDN_DOCKER_CONTAINER_LISTING)))
+        end
+
+        # eg content url listing file ->
+        # /content/dist/rhel/server/7/7Server/x86_64/containers/CONTAINER_REGISTRY_LISTING
+        # format
+        #   {
+        #   "header": {
+        #       "version": "1.0"
+        #   },
+        #   "payload": {
+        #       "registries": [
+        #           { "name": "rhel",
+        #             "url": "<url>",
+        #             },
+        #           { "name": "rhel7",
+        #             "url": "<url>",
+        #             "aliases": [ "redhat/rhel7" ]
+        #             }
+        #       ]
+        #   }
+        # }
+        def get_docker_registries(content_path)
+          docker_listing = get_container_listings(content_path)
+          if docker_listing["payload"]
+            docker_listing["payload"]["registries"] || []
```

Comment:
My concern is that it's not easy to read because you have two checks. Would be nice to combine them into one. However you want to do that is fine by me.